gtk4/src/subclass/entry_buffer.rs

```diff
@@ -0,0 +1,233 @@
+use crate::EntryBuffer;
+use glib::subclass::object::ObjectImpl;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, GString, Object};
+
+pub trait EntryBufferImpl: EntryBufferImplExt + ObjectImpl {
+    fn delete_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) -> u32 {
+        self.parent_delete_text(entry_buffer, position, n_chars)
+    }
+
+    fn deleted_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) {
+        self.parent_deleted_text(entry_buffer, position, n_chars)
+    }
+
+    fn get_length(&self, entry_buffer: &Self::Type) -> u32 {
+        self.parent_get_length(entry_buffer)
+    }
+
+    fn get_text(&self, entry_buffer: &Self::Type, n_bytes: &mut usize) -> GString {
+        self.parent_get_text(entry_buffer, n_bytes)
+    }
+    fn insert_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str) -> u32 {
+        self.parent_insert_text(entry_buffer, position, chars)
+    }
+
+    fn inserted_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str) {
+        self.parent_inserted_text(entry_buffer, position, chars)
+    }
+}
+
+pub trait EntryBufferImplExt: ObjectSubclass {
+    fn parent_delete_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) -> u32;
+    fn parent_deleted_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32);
+    fn parent_get_length(&self, entry_buffer: &Self::Type) -> u32;
+    fn parent_get_text(&self, entry_buffer: &Self::Type, n_bytes: &mut usize) -> GString;
```

Comment:
This is all a bit suboptimal as you'd always create multiple copies of the string here on the way out. This needs to be refactored a bit later (create an issue please).

The idea of the C class is that you return a plain borrowed reference to the string buffer here without any copying, but I'm not entirely sure how that can be safely done here. Typical C-isms in GTK like all those `transfer none` return values there and praying that nothing explodes.

Reply:
I think this one is handled as well now?

Reply:
Well it's still suboptimal, there are many copies here :) One for the `to_glib_full()`, one for getting the return value from the parent class' impl, another for getting the value from the Rust impl.